### PR TITLE
Represent segments/checkpoints in Python, use in status CLI

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Preparation for #4996 to make that PR a bit smaller.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by nilsvu** [comment]
@wthrowe @knelli2 we can do this subset of #4996 first to make the other PR a bit smaller.

File: tools/Status/Status.py [hunk 1]
```diff
@@ -224,6 +233,34 @@ def render_status(show_paths, show_unidentified, state_styles, **kwargs):
     if len(job_data) == 0:
         return
 
+    # Remove deleted jobs
+    if not show_deleted:
+        deleted_jobs = job_data[~job_data["WorkDir"].map(os.path.exists)]
+        job_data.drop(deleted_jobs.index, inplace=True)
+
+    # Keep only latest in a series of segments
+    job_data[["SegmentsDir", "SegmentId"]] = [
```
Comment (knelli2) [inline tools/Status/Status.py:242]: Can we add a column that prints what segment we are on? I think that could be useful. Then we don't have to include it in the job name like SpEC does.

**Review by knelli2 — CHANGES_REQUESTED** [review]
Should you include the changes to Main that actually change the checkpoint structure in this PR?

File: tools/Status/Status.py (hunk 1 above)
Comment (nilsvu) [reply tools/Status/Status.py:242]: I had that before but didn't find it particularly useful. I can add it if you want though. Not sure.

Comment (knelli2) [reply tools/Status/Status.py:242]: For long runs I think it'd be useful. It's a very quick way to know how many times the simulation has restarted.

**Comment by nilsvu** [comment]
Moved the changes to Main to #5079

File: tools/Status/Status.py (hunk 1 above)
Comment (nilsvu) [reply tools/Status/Status.py:242]: Right but is it useful to know the number of restarts? I don't know. Remember we have limited space in the terminal, so adding a column means things like the job name get truncated further.

Comment (knelli2) [reply tools/Status/Status.py:242]: I feel like it should be an option. Everybody is going to want to see different things, so maybe (for a later PR) we should take the fields to print as arguments so users can customize them in their spectre.yaml file

Comment (nilsvu) [reply tools/Status/Status.py:242]: Yea that would be nice

**Comment by nilsvu** [comment]
@wthrowe could you take this?

File: tests/support/Python/Test_DirectoryStructure.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    list_segments,
+)
+
+
+class TestDirectoryStructure(unittest.TestCase):
+    def setUp(self):
+        self.test_dir = Path(
+            unit_test_build_path(), "support/DirectoryStructure"
+        )
+        self.test_dir.mkdir(parents=True, exist_ok=True)
```
Comment (wthrowe) [inline tests/support/Python/Test_DirectoryStructure.py:24]: Delete any preexisting files.

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: tests/support/Python/Test_DirectoryStructure.py (hunk 2 above)
Comment (nilsvu) [reply tests/support/Python/Test_DirectoryStructure.py:24]: ok fixed

**Review by wthrowe — APPROVED** [review]

**Comment by nilsvu** [comment]
@wthrowe I think this is ready to go as well. @knelli2 is on vacation this week and we can make any changes to the `status` CLI in follow-ups if we want to.

**Comment by wthrowe** [comment]
Agreed.  I don't see @knelli2's request as blocking.

**Pull request merged**